packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -0,0 +1,1107 @@
+  controlPointAdditionDistance: 6,
+  controlPointDeletionDistance: 6,
+  showControlPointsConnectors: false,
+  controlPointAdditionEnabled: true,
+  controlPointDeletionEnabled: true,
+};
+
+class SplineROITool extends AnnotationTool {
+  static toolName;
+
+  touchDragCallback: any;
+  mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  editData: {
+    annotation: SplineROIAnnotation;
+    viewportIdsToRender: Array<string>;
+    handleIndex?: number;
+    movingTextBox?: boolean;
+    newAnnotation?: boolean;
+    hasMoved?: boolean;
+  } | null;
+  isDrawing: boolean;
+  isHandleOutsideImage = false;
+  splineInstancesCache: WeakMap<SplineROIAnnotation, ISpline> = new Map();
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        preventHandleOutsideImage: false,
+        calculateStats: true,
+        getTextLines: defaultGetTextLines,
+        spline: {
+          configuration: {
+            CARDINAL: {
+              Class: CardinalSpline,
+              scale: 0.5,
+            },
+            CATMULLROM: {
+              Class: CatmullRomSpline,
+            },
+            LINEAR: {
+              Class: LinearSpline,
+            },
+            BSPLINE: {
+              Class: BSpline,
+              controlPointAdditionEnabled: false,
+              controlPointDeletionEnabled: false,
+              showControlPointsConnectors: true,
+            },
+          },
+          type: 'CATMULLROM',
+        },
+        actions: [
+          {
+            method: 'addControlPointCallback',
+            bindings: [
+              {
+                mouseButton: MouseBindings.Primary,
+                modifierKey: KeyboardBindings.Shift,
+              },
+            ],
+          },
+          {
+            method: 'deleteControlPointCallback',
+            bindings: [
+              {
+                mouseButton: MouseBindings.Primary,
+                modifierKey: KeyboardBindings.Ctrl,
+              },
+            ],
+          },
+        ],
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+
+    this._throttledCalculateCachedStats = throttle(
+      this._calculateCachedStats,
+      100,
+      { trailing: true }
+    );
+  }
+
+  /**
+   * Based on the current position of the mouse and the current imageId to create
+   * a CircleROI Annotation and stores it in the annotationManager
+   *
+   * @param evt -  EventTypes.NormalizedMouseEventType
+   * @returns The annotation object.
+   *
+   */
+  addNewAnnotation = (
+    evt: EventTypes.InteractionEventType
+  ): SplineROIAnnotation => {
+    const eventDetail = evt.detail;
+    const { currentPoints, element } = eventDetail;
+    const worldPos = currentPoints.world;
+
+    const enabledElement = getEnabledElement(element);
+    const { viewport, renderingEngine } = enabledElement;
+
+    this.isDrawing = true;
```

Comment:
It is odd that we have `isDrawing = true` at the top. Usually, other tools place it at the bottom.

Reply:
The this.isDrawing is usually the last thing (or close to last) in the addNewAnnotation in any tools